src/decryptor.rs

```diff
@@ -0,0 +1,53 @@
+use sdkms::{
+    api_model::{Algorithm::Rsa, DecryptRequest, SobjectDescriptor},
+    SdkmsClient,
+};
+use sequoia_openpgp::{
+    crypto::{mpi, Decryptor, SessionKey},
+    packet::{
+        key::{PublicParts, UnspecifiedRole},
+        Key,
+    },
+    Result as SequoiaResult,
+};
+
+pub(crate) struct RawDecryptor<'a> {
+    pub(crate) api_endpoint: &'a str,
+    pub(crate) api_key:      &'a str,
+    pub(crate) descriptor:   &'a SobjectDescriptor,
+    pub(crate) public:       &'a Key<PublicParts, UnspecifiedRole>,
+}
```

Comment:
I'd be inclined to make these field `pub` instead of `pub(crate)`, since the struct itself is already confined to the crate